apps/storybook/components/funke/IdCardPinScreen.stories.tsx

```diff
@@ -0,0 +1,243 @@
+import {
+  Button,
+  Heading,
+  type HeroIcon,
+  HeroIcons,
+  Image,
+  Page,
+  Paragraph,
+  ProgressBar,
+  Stack,
+  XStack,
+  YStack,
+} from '@package/ui'
+import type { Meta, StoryObj } from '@storybook/react'
+import React, { useRef, useState } from 'react'
+import { StyleSheet, type TextInput } from 'react-native'
+import { Circle, Input } from 'tamagui'
+import { LinearGradient } from 'tamagui/linear-gradient'
+
+import { useArgs } from '@storybook/addons'
+
+const germanIssuerImage = require('../../../funke/assets/german-issuer-image.png')
+
+interface IdCardPinScreenProps {
+  pinLength: number
+  onPinComplete: (pin: string) => void
+  state: 'enterPin' | 'loading' | 'complete'
+  onGoToWallet: () => void
+  userName?: string
+}
+
+const stateMapping = {
+  enterPin: { title: 'Enter passport pin', icon: 'locked' },
+  loading: { title: 'Setting up identity', icon: 'loading' },
+  complete: { title: 'You wallet is ready', icon: 'complete' },
+} as const
+
+const iconMapping = {
+  locked: <HeroIcons.LockClosed color="$white" />,
+  loading: <HeroIcons.ArrowPath color="$white" />,
+  complete: <HeroIcons.ShieldCheck color="$white" />,
+} as const
+
+interface IdCardProps {
+  icon: keyof typeof iconMapping
+  issuerImage: string
+  userName?: string
+}
+
+function IdCard({ icon, issuerImage, userName }: IdCardProps) {
+  return (
+    <YStack gap="$6" p="$5" borderRadius="$8" overflow="hidden" borderWidth={1} borderColor="rgba(216, 218, 200, 1)">
+      <LinearGradient
+        colors={['#EFE7DA', '#EDEEE6', '#E9EDEE', '#D4D6C0']}
+        start={[0.98, 0.02]}
+        end={[0.28, 1.0]}
+        locations={[0.0207, 0.3341, 0.5887, 1.0]}
+        style={StyleSheet.absoluteFillObject}
+      />
+      <LinearGradient
+        colors={['rgba(255, 255, 255, 0.2)', 'rgba(255, 255, 255, 0.2)']}
+        start={[0, 0]}
+        end={[1, 0]}
+        style={StyleSheet.absoluteFillObject}
+      />
+      <XStack justifyContent="space-between">
+        <YStack gap="$1">
+          <Paragraph secondary>Personalausweis</Paragraph>
+          <Paragraph size="$6" fontWeight="$regular">
+            {userName ?? '********'}
+          </Paragraph>
+        </YStack>
+        <Stack>
+          <Image src={issuerImage} width={75} height={75} resizeMode="contain" />
+        </Stack>
+      </XStack>
+      <XStack justifyContent="flex-end">
+        {/* TODO: background color variables */}
+        <Circle size="$3" backgroundColor={icon === 'complete' ? '#C9D6BD' : '#02010033'}>
+          {iconMapping[icon]}
```

Comment:
Might be okay for now to just have a grey circle with a slight opacity so that it blends a bit with where we put it. (no need for colours)